native/cocos/core/Root.cpp

```diff
@@ -71,10 +75,52 @@ Root::~Root() {
     instance = nullptr;
 }
 
-void Root::initialize(gfx::Swapchain *swapchain) {
-    _swapchain = swapchain;
+void Root::initialize(gfx::Swapchain *  /*swapchain*/) {
+    auto *windowMgr = CC_GET_PLATFORM_INTERFACE(ISystemWindowManager);
+    const auto &windows = windowMgr->getWindows();
+    for (const auto &pair : windows) {
+        auto *window = pair.second.get();
+        scene::RenderWindow *renderWindow = createRenderWindowFromSystemWindow(window);
+        if (!_mainRenderWindow) {
+            _mainRenderWindow = renderWindow;
```

Comment:
How to make sure _mainRenderWindow is created with main system window?

Reply:
Maybe change to `if (!_mainRenderWindow && (window->getWindowId() == 1))` ?

Reply:
Yep, I think so.